core/src/blocktree.rs

```diff
@@ -921,6 +966,41 @@ impl Blocktree {
         batch_processor.write(batch)?;
         Ok(())
     }
+
+    /// Prune blocktree such that slots higher than `target_slot` are deleted and all references to
+    /// higher slots are removed
+    pub fn prune(&self, target_slot: u64) {
+        let mut meta = self
+            .meta(target_slot)
+            .expect("couldn't read slot meta")
+            .expect("no meta for root slot");
+        meta.next_slots.clear();
+        self.put_meta_bytes(
+            target_slot,
+            &bincode::serialize(&meta).expect("couldn't get meta bytes"),
+        )
+        .expect("unable to update meta");
+
+        self.slot_meta_iterator(target_slot + 1)
+            .expect("unable to iterate over meta")
+            .for_each(|(slot, _)| self.delete_slot(slot).unwrap());
```

Comment:
ok, errors thrown to this point.  how about an expect() instead?

Reply:
this code is going to be in heavy use in the near term, and we don't know what errors are gonna pop out, we need to log the hell out of it in order to best support the community

Reply:
Yeah agreed, I think I rushed into marking this as ready. Was just glad it even worked lol